### ISSUE TITLE
Pilot: Document all annotations used

### ISSUE THREAD

**Issue by geeknoid** [istio/istio#12940]
Need to document all annotations used in the code. Search for annotations.Register and fill in the necessary description string. These descriptions will go to istio.io as end-user docs.

**Comment by esnible** [comment]
I wish to start putting new annotations into a well-known place in preparation for this item.  _pkg/annotations/annotation.go_ already exists.  Is it reasonable?  (May I create _pkg/annotations/constants.go_ or similar if not?)

**Comment by geeknoid** [comment]
No, I don't think it's the right idea to put all the known annotations in
the otherwise completely generic package designed to track any annotation.
Further, I don't think we want a centralized list, since it breaks
encapsulation of individual componenta.

If you look at how the annotations are handled in the sidecar injector, all
of those are together in one table for that specific component.

**Comment by esnible** [comment]
@geeknoid I'm thinking about https://github.com/istio/istio.io/issues/3191 .  I just created a PR, https://github.com/istio/istio/pull/13092 , with two brand-new annotations at the request of @mandarjog .  What do I need to do to get these annotations documented on istio.io?

My dream is that istio.io will have a single page with all of the annotations so that if a user sees an unrecognized annotation they know where to go to look it up.  Failing that I need the annotations to at least be searchable.  Currently if I search for `sidecar.istio.io/statsInclusionRegexps` no results are returned.

**Comment by geeknoid** [comment]
I've put in the infrastructure needed to have all annotation usage be
documented directly in the code. Every use of an annotation ends up calling
pkg/annotations/Register() which gives the name of the annotation and a
description. The set of annotations consumed by each command is shown in
the command's page (so the 'pilot' page or the 'galley' page). There is a
similar mechanism in place now to track and document environment variable
use by individual commands.

Although I've added the necessary registration calls for annotations and
environment variables, I didn't add any descriptions. Instead, I've opened
GitHub issues for the component owners to go and add the requisite
descriptions.

I checked this stuff in just a few weeks ago in master, so it's not in the
1.1 branch.
